src/charm.py

```diff
@@ -151,7 +138,16 @@ def _on_install(self, event):
         """Event handler for InstallEvent."""
 
         # Update Pebble configuration layer if it has changed
-        self._update_layer()
+        try:
+            update_layer(
+                self._name, self._container, self._training_operator_layer, logger
+            )
+        except ErrorWithStatus as e:
+            self.model.unit.status = e.status
+            if isinstance(e.status, BlockedStatus):
+                self.logger.error(str(e.msg))
+            else:
+                self.logger.info(str(e.msg))
```

Comment:
Let's better use `logger.error` if we are logging the message from an exception.